01_fib/ovlords/main.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func fib(n int) {
+
+	if n > 0 {
+
+		currentNumber, previousNumber, scratchNumber := 1, 0, 0
+
+		for lines := 1; lines <= n; lines++ {
+			fmt.Fprintln(out, currentNumber)
+
+			scratchNumber, previousNumber = previousNumber, currentNumber
+			currentNumber += scratchNumber
+		}
+
+	} else {
+		fmt.Fprintln(out, "Invalid Input: Must be a positive integer")
```

Comment:
A very common Go idiom is "early return". You check the inputs up front and if invalid, you return at the top of the function. This allows the other branch of the body to be outdented a level.

See https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88 for a blog article about this style and why it is done.